skbio/embedding/_protein.py

```diff
@@ -0,0 +1,97 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+from skbio.sequence import Protein
+from skbio.embedding._embedding import SequenceEmbedding
+from skbio.util import get_data_path
+from pathlib import Path
+import pandas as pd
+import numpy as np
+
+
+class ProteinEmbedding(SequenceEmbedding):
+    """Stores the embeddings of the protein sequence.
+
+    Parameters
+    ----------
+    sequence : str, Sequence, or 1D np.ndarray
+        Characters representing the protein sequence itself.
+    embedding : np.ndarray
+        The embedding of the protein sequence. Row vectors correspond to
+        the latent residues coordinates.
+    clip_head : bool, optional
+        If ``True``, then the first row of the embedding will be removed.
+        Some language models specify start tokens, and this parameter can
+        be used to account for this.
+    clip_end : bool, optional
+        If ``True``, then the last row of the embedding will be removed.
+        Some language models specify end tokens, and this parameter can
+        be used to account for this.
+
+    See Also
+    --------
+    Protein
+
+    """
+
+    default_write_format = "embed"
+
+    def __init__(
+        self, embedding, sequence: str, clip_head=False, clip_tail=False, **kwargs
+    ):
+        if clip_head:
+            embedding = embedding[1:]
+        if clip_tail:
+            embedding = embedding[:-1]
+
+        if isinstance(sequence, Protein):
```

Comment:
I'm not clear on this, how exactly would references in python be stored? Python doesn't have explicit support for pointers AFAIK

In this case, this is a minimal memory footprint, since the vast majority of proteins have less than 1K residues (and consequentially the vast majority of pLMs can only handle proteins with less than 1K residues). Plus the embeddings here are by far takes up most the memory, for each residue, there is a vector of length 1024 for ProTrans

I'm OK with having the internal sequence attribute be a protein.  Now the question is, does it make more sense for `ProteinEmbedding` to have a protein attribute, or for it to subclass `Protein`?

Reply:
I think that in Python, all variables are [passed by reference](https://realpython.com/python-pass-by-reference/), therefore you don't need to explicitly define a pointer. It is done already. If you do need to pass by value, you will have to manually trigger `.copy` or `.deepcopy` or something.

I would suggest that a `SequenceEmbedding` class has a `sequence` attribute. For each subclass (`ProteinEmbedding`, `DNAEmbedding`, `RNAEmbedding`), we can have explicit constraints (e.g., `sequence` must be `Protein`) and checks.

Reply:
PS: Even if you do:
```python
prot = Protein('MKRVGI')
emb = ProteinEmbedding(..., prot)
prot = None
```
The embedding object still has the protein information. I think.